scripts/mod_loader/modapi/map.lua

```diff
@@ -15,6 +15,19 @@ function modApi:fileExists(name)
 	end
 end
 
+function modApi:directoryExists(path)
+	if type(path) ~= "string" then return false end
+
+	local response = os.execute("cd " .. path)
```

Comment:
I know I used it myself, but... `os.execute` causes the command line terminal to appear briefly, which is not only ugly and startling for non-tech savvy players, but also somewhat slow, since the operating system has to start up a new terminal to execute the command. 
Generally I'd advise against using it in API functions - rather, limit its usage to internal mod loader and game startup only, if it's absolutely required.

Luckily, it's possible to use another function from `os` here: `os.rename` ([see example](https://stackoverflow.com/a/40195356)), which doesn't cause the terminal to pop up and is *much* faster.

`modApi:fileExists` will also benefit from this, since we won't have to try to open the file for reading.

Reply:
Ah, I was not aware. That would indeed be ugly.

With that change; `fileExists` will return true for both files and folders, while `directoryExists` will return true only for folders.
Would you consider that correct behavior?

Reply:
Yes, that'd be correct, given the tools we have to work with.